recipe/build.sh

```diff
@@ -1,25 +1,50 @@
 #!/bin/bash
 
+set -e
+# Stuart's recommendation to stop lapack-test from failing
+ulimit -s 50000
+
+# https://github.com/xianyi/OpenBLAS/wiki/faq#Linux_SEGFAULT
+patch < segfaults.patch
+
 # See this workaround
 # ( https://github.com/xianyi/OpenBLAS/issues/818#issuecomment-207365134 ).
-CF="${CFLAGS}"
+export CF="${CPPFLAGS} ${CFLAGS} -fopenmp -Wno-unused-parameter -Wno-old-style-declaration"
 unset CFLAGS
+export LAPACK_FFLAGS="${FFLAGS}"
 
 if [[ `uname` == 'Darwin' ]]; then
      export LIBRARY_SEARCH_VAR=DYLD_FALLBACK_LIBRARY_PATH
+     export CF="$CF -Wl,-rpath,$PREFIX/lib"
+     export LAPACK_FFLAGS="$LAPACK_FFLAGS -Wl,-rpath,$PREFIX/lib"
+     export FFLAGS="$FFLAGS -Wl,-rpath,$PREFIX/lib"
+     USE_OPENMP="0"
 else
      export LIBRARY_SEARCH_VAR=LD_LIBRARY_PATH
+    # Gnu OpenMP is not fork-safe.  We disable openmp right now, so that downstream packages don't hang as a result of this.
+    # USE_OPENMP="1"
+    USE_OPENMP="0"
 fi
 eval export ${LIBRARY_SEARCH_VAR}="${PREFIX}/lib"
 
+
+# USE_SIMPLE_THREADED_LEVEL3 is necessary to avoid hangs when more than one process uses blas:
+#    https://github.com/xianyi/OpenBLAS/issues/1456
+#    https://github.com/xianyi/OpenBLAS/issues/294
+#    https://github.com/scikit-learn/scikit-learn/issues/636
+
+
 # Build all CPU targets and allow dynamic configuration
 # Build LAPACK.
 # Enable threading. This can be controlled to a certain number by
 # setting OPENBLAS_NUM_THREADS before loading the library.
-make DYNAMIC_ARCH=1 BINARY=${ARCH} NO_LAPACK=0 NO_AFFINITY=1 USE_THREAD=1 NUM_THREADS=128 CFLAGS="${CF}" FFLAGS="-frecursive"
-OPENBLAS_NUM_THREADS=$CPU_COUNT make test
+make DYNAMIC_ARCH=1 BINARY=${ARCH} NO_LAPACK=0 NO_AFFINITY=1 USE_THREAD=1 NUM_THREADS=128 \
+     USE_OPENMP="${USE_OPENMP}" USE_SIMPLE_THREADED_LEVEL3=1 CFLAGS="${CF}" FFLAGS="${FFLAGS}"
+OPENBLAS_NUM_THREADS=$CPU_COUNT CFLAGS="${CF}" FFLAGS="${FFLAGS}" make test
+CFLAGS="${CF}" FFLAGS="${FFLAGS}"
```

Comment:
Are these environment variables suppose to go along with something else (i.e. a command)?

Reply:
no this should be removed